packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -155,23 +219,50 @@ export const useArbTokenBridge = (
     }
   }, [walletAddress, bridge])
 
-  const depositEth = async (weiValue: BigNumber) => {
-    const etherVal = utils.formatUnits(weiValue, 18)
-    const tx = await bridge.depositETH(weiValue)
+  const depositEth = async (amount: BigNumber) => {
+    // TODO: clean up
+    if (typeof l1Signer === 'undefined') {
+      throw new Error(`No instance of L1Signer found`)
+    }
+
+    if (typeof l2Signer === 'undefined') {
+      throw new Error(`No instance of L2Signer found`)
+    }
+
+    if (typeof l2Signer.provider === 'undefined') {
+      throw new Error(`No instance of L2Signer found`)
+    }
+
+    if (typeof ethBridger === 'undefined') {
+      throw new Error(`No instance of EthBridger found`)
+    }
+
+    const tx = await ethBridger.deposit({
+      l1Signer,
+      l2Provider: l2Signer.provider,
+      amount
+    })
+
     addTransaction({
       type: 'deposit-l1',
       status: 'pending',
-      value: etherVal,
+      value: utils.formatUnits(amount, 18),
       txID: tx.hash,
       assetName: 'ETH',
       assetType: AssetType.ETH,
       sender: await walletAddressCached(),
       l1NetworkID: await l1NetworkIDCached()
     })
+
     const receipt = await tx.wait()
-    const seqNums = await bridge.getInboxSeqNumFromContractTransaction(receipt)
-    if (!seqNums) return
-    const seqNum = seqNums[0]
+    const messages = await receipt.getL1ToL2Messages(l2Signer)
+
+    if (messages.length === 0) {
```

Comment:
on the following lines we kinda assume there is only a single result, we could enforce it here (and maybe log if this expectation is broken)
```suggestion
    if (messages.length !== 1) {
```